float32[3] rotation
int32 throttle
float32 depth